### PR TITLE
[MOB-2979] Analytics integration tests continuation

### PR DESCRIPTION
**Pull request by d4r1091**
[MOB-2979]

## Context

We started and implemented the first batch of Analytics integration tests.
This PR wants to enrich its content of.

If you want to utilize this PR and finish all the remaining ones, can do 👍 

## Approach

Some "accessors stretch" was made to accommodate the testing.
Despite some enums were `CaseIterable`, I was unable to access to the `allCases` so I ended up re-adding them into the tests.

Added the following tests

- "Learn" more tap on the About Ecosia section
- Top site actions
- Multiply Impact screens (referrals)
- News items and detail screen view
- The whole onboarding flow

> For the nasty issue with `linkCopying` test, I realized there is something wrong with the `.touchUpdInside` event only when run in CI. I removed it for now as we may find another solution later. I didn't want to keep the test but skipping it as it could be easily missed so when continuing with Analytics Integration Test we can find it still as "to to".
My suggestion is to review that component altogether and make it as a UIButton so that we can guarantee reliability. If you agree I can make a ticket out of it.

## Other

- It also aims to standardize the comments throughout the testing functions
- Fixes some linting issues

### Checklist

- [x] I performed some relevant testing on a real device and/or simulator
- [x] I wrote Unit Tests that confirm the expected behavior

[MOB-2979]: https://ecosia.atlassian.net/browse/MOB-2979?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: EcosiaTests/Analytics/AnalyticsSpyTests.swift [hunk 1]
```diff
@@ -33,8 +39,13 @@ final class AnalyticsSpy: Analytics {
     }
 
     var menuClickItemCalled: Analytics.Label.Menu?
+    var menuClickExpectation: XCTestExpectation?
     override func menuClick(_ item: Analytics.Label.Menu) {
         menuClickItemCalled = item
+        DispatchQueue.main.async {
+            self.menuClickExpectation?.fulfill()
+            self.menuClickExpectation = nil // Prevent multiple fulfillments
+        }
```
Comment (lucaschifino) [inline EcosiaTests/Analytics/AnalyticsSpyTests.swift:48]: Thank you for handling the flakyness of the menu tests, it's indeed quite tricky 🙏 

Can you elaborate why setting this to nil is needed? Shouldn't we expect this to be called only once per test?

**Review by lucaschifino — DISMISSED** [review]
Thank you for the tests! No blockers from my side, just two clarifications

File: EcosiaTests/Analytics/AnalyticsSpyTests.swift (hunk 1 above)
Comment (d4r1091) [reply EcosiaTests/Analytics/AnalyticsSpyTests.swift]: Tbh there wasn't much of a need for them. Once I solved the problem at the root I should have removed them. Did it now. Thanks! 😉

**Review by lucaschifino — APPROVED** [review]
Nice, thanks! 👏

**Pull request merged**